### ISSUE TITLE
Fix compiler warnings in MSVC 2013 64 bit

### ISSUE THREAD

**Issue by acki-m** [libnonius/nonius#9]
I get a lot of compiler warning while compiling the 64 bit version of nonius.
Version: 1.0.0

e.g. something like this:

```
int quotes = std::count(first, last, '"');
```

Full MSVC 2013 64 bit output:

```
.../nonius/reporters/csv_reporter.h++(102): warning C4244: 'initializing' : conversion from '__int64' to 'int', possible loss of data
1>.../nonius/detail/escape.h++(32): warning C4244: 'initializing' : conversion from '__int64' to 'int', possible loss of data
1>.../nonius/reporters/junit_reporter.h++(94): warning C4244: 'initializing' : conversion from '__int64' to 'int', possible loss of data
1>.../nonius/detail/argparse.h++(73): warning C4267: 'return' : conversion from 'size_t' to 'int', possible loss of data
1>          .../nonius/detail/argparse.h++(79) : see reference to function template instantiation 'int nonius::detail::get_max_width<std::_Vector_const_iterator<std::_Vector_val<std::_Simple_types<nonius::detail::option>>>,nonius::detail::<<::<lambda_ebcb31a12dfc7293f4c5c592439004ad>>(Iterator,Iterator,Projection)' being compiled
1>          with
1>          [
1>              Iterator=std::_Vector_const_iterator<std::_Vector_val<std::_Simple_types<nonius::detail::option>>>
1>  ,            Projection=nonius::detail::<<::<lambda_ebcb31a12dfc7293f4c5c592439004ad>
1>          ]
1>.../nonius/detail/stats.h++(199): warning C4244: 'initializing' : conversion from '__int64' to 'int', possible loss of data
1>          .../nonius/detail/analyse.h++(33) : see reference to function template instantiation 'nonius::detail::bootstrap_analysis nonius::detail::analyse_samples<std::_Vector_iterator<std::_Vector_val<std::_Simple_types<double>>>>(double,int,Iterator,Iterator)' being compiled
1>          with
1>          [
1>              Iterator=std::_Vector_iterator<std::_Vector_val<std::_Simple_types<double>>>
1>          ]
1>          .../nonius/go.h++(88) : see reference to function template instantiation 'nonius::sample_analysis<boost::chrono::duration<double,boost::ratio<0x01,0x03b9aca00>>> nonius::detail::analyse<boost::chrono::duration<double,boost::ratio<0x01,0x03b9aca00>>,std::_Vector_iterator<std::_Vector_val<std::_Simple_types<boost::chrono::duration<double,boost::ratio<0x01,0x03b9aca00>>>>>>(nonius::configuration,nonius::environment<boost::chrono::duration<double,boost::ratio<0x01,0x03b9aca00>>>,Iterator,Iterator)' being compiled
1>          with
1>          [
1>              Iterator=std::_Vector_iterator<std::_Vector_val<std::_Simple_types<boost::chrono::duration<double,boost::ratio<0x01,0x03b9aca00>>>>>
1>          ]
1>          .../nonius/go.h++(130) : see reference to function template instantiation 'void nonius::go<nonius::default_clock,std::_Vector_iterator<std::_Vector_val<std::_Simple_types<nonius::benchmark>>>>(nonius::configuration,Iterator,Iterator,nonius::reporter &)' being compiled
1>          with
1>          [
1>              Iterator=std::_Vector_iterator<std::_Vector_val<std::_Simple_types<nonius::benchmark>>>
1>          ]
1>          .../nonius/main.h++(145) : see reference to function template instantiation 'void nonius::go<nonius::default_clock>(nonius::configuration,nonius::benchmark_registry &,nonius::reporter_registry &)' being compiled
1>.../nonius/detail/stats.h++(80): warning C4244: 'initializing' : conversion from '__int64' to 'int', possible loss of data
1>          .../nonius/detail/estimate_clock.h++(64) : see reference to function template instantiation 'double nonius::detail::mean<std::_Vector_iterator<std::_Vector_val<std::_Simple_types<double>>>>(Iterator,Iterator)' being compiled
1>          with
1>          [
1>              Iterator=std::_Vector_iterator<std::_Vector_val<std::_Simple_types<double>>>
1>          ]
1>          .../nonius/go.h++(42) : see reference to function template instantiation 'nonius::environment_estimate<boost::chrono::duration<double,boost::ratio<0x01,0x03b9aca00>>> nonius::detail::estimate_clock_resolution<Clock>(int)' being compiled
1>          with
1>          [
1>              Clock=nonius::default_clock
1>          ]
1>          .../nonius/go.h++(73) : see reference to function template instantiation 'nonius::environment<boost::chrono::duration<double,boost::ratio<0x01,0x03b9aca00>>> nonius::detail::measure_environment<Clock>(nonius::reporter &)' being compiled
1>          with
1>          [
1>              Clock=nonius::default_clock
1>          ]
1>.../nonius/detail/stats.h++(42): warning C4244: 'initializing' : conversion from '__int64' to 'int', possible loss of data
1>          .../nonius/detail/stats.h++(58) : see reference to function template instantiation 'double nonius::detail::weighted_average_quantile<std::_Vector_iterator<std::_Vector_val<std::_Simple_types<double>>>>(int,int,Iterator,Iterator)' being compiled
1>          with
1>          [
1>              Iterator=std::_Vector_iterator<std::_Vector_val<std::_Simple_types<double>>>
1>          ]
1>          .../nonius/detail/analyse.h++(34) : see reference to function template instantiation 'nonius::outlier_classification nonius::detail::classify_outliers<std::_Vector_iterator<std::_Vector_val<std::_Simple_types<double>>>>(Iterator,Iterator)' being compiled
1>          with
1>          [
1>              Iterator=std::_Vector_iterator<std::_Vector_val<std::_Simple_types<double>>>
1>          ]
1>.../nonius/detail/stats.h++(97): warning C4244: 'initializing' : conversion from '__int64' to 'int', possible loss of data
1>          .../nonius/detail/stats.h++(208) : see reference to function template instantiation 'nonius::detail::sample nonius::detail::resample<std::mt19937,Iterator,double(__cdecl *const )(Iterator,Iterator)>(URng &,int,Iterator,Iterator,Estimator &)' being compiled
1>          with
1>          [
1>              Iterator=std::_Vector_iterator<std::_Vector_val<std::_Simple_types<double>>>
1>  ,            URng=std::mt19937
1>  ,            Estimator=double (__cdecl *const )(std::_Vector_iterator<std::_Vector_val<std::_Simple_types<double>>>,std::_Vector_iterator<std::_Vector_val<std::_Simple_types<double>>>)
1>          ]
1>          D:\Programmierung\IDE\Microsoft Visual Studio 12.0\VC\include\xmemory0(577) : while compiling class template member function 'std::_List_node<std::pair<const _Kty,_Ty>,void *> *std::allocator<_Other>::allocate(std::allocator<_Other>::size_type)'
1>          with
1>          [
1>              _Kty=std::string
1>  ,            _Ty=std::string
1>  ,            _Other=std::_List_node<std::pair<const std::string,std::string>,void *>
1>          ]
1>          D:\Programmierung\IDE\Microsoft Visual Studio 12.0\VC\include\xmemory0(847) : see reference to function template instantiation 'std::_List_node<std::pair<const _Kty,_Ty>,void *> *std::allocator<_Other>::allocate(std::allocator<_Other>::size_type)' being compiled
1>          with
1>          [
1>              _Kty=std::string
1>  ,            _Ty=std::string
1>  ,            _Other=std::_List_node<std::pair<const std::string,std::string>,void *>
1>          ]
1>          D:\Programmierung\IDE\Microsoft Visual Studio 12.0\VC\include\xmemory0(749) : see reference to class template instantiation 'std::allocator<_Other>' being compiled
1>          with
1>          [
1>              _Other=std::_List_node<std::pair<const std::string,std::string>,void *>
1>          ]
1>          D:\Programmierung\IDE\Microsoft Visual Studio 12.0\VC\include\list(484) : see reference to class template instantiation 'std::_Wrap_alloc<std::allocator<_Other>>' being compiled
1>          with
1>          [
1>              _Other=std::_List_node<std::pair<const std::string,std::string>,void *>
1>          ]
1>          D:\Programmierung\IDE\Microsoft Visual Studio 12.0\VC\include\list(690) : see reference to class template instantiation 'std::_List_base_types<_Ty,_Alloc>' being compiled
1>          with
1>          [
1>              _Ty=std::pair<const std::string,std::string>
1>  ,            _Alloc=std::allocator<std::pair<const std::string,std::string>>
1>          ]
1>          D:\Programmierung\IDE\Microsoft Visual Studio 12.0\VC\include\list(814) : see reference to class template instantiation 'std::_List_alloc<false,std::_List_base_types<_Ty,_Alloc>>' being compiled
1>          with
1>          [
1>              _Ty=std::pair<const std::string,std::string>
1>  ,            _Alloc=std::allocator<std::pair<const std::string,std::string>>
1>          ]
1>          D:\Programmierung\IDE\Microsoft Visual Studio 12.0\VC\include\list(862) : see reference to class template instantiation 'std::_List_buy<_Ty,_Alloc>' being compiled
1>          with
1>          [
1>              _Ty=std::pair<const std::string,std::string>
1>  ,            _Alloc=std::allocator<std::pair<const std::string,std::string>>
1>          ]
1>          D:\Programmierung\IDE\Microsoft Visual Studio 12.0\VC\include\xhash(269) : see reference to class template instantiation 'std::list<std::pair<const _Kty,_Ty>,std::allocator<std::pair<const _Kty,_Ty>>>' being compiled
1>          with
1>          [
1>              _Kty=std::string
1>  ,            _Ty=std::string
1>          ]
1>          D:\Programmierung\IDE\Microsoft Visual Studio 12.0\VC\include\unordered_map(81) : see reference to class template instantiation 'std::_Hash<std::_Umap_traits<_Kty,_Ty,std::_Uhash_compare<_Kty,_Hasher,_Keyeq>,_Alloc,false>>' being compiled
1>          with
1>          [
1>              _Kty=std::string
1>  ,            _Ty=std::string
1>  ,            _Hasher=std::hash<std::string>
1>  ,            _Keyeq=std::equal_to<std::string>
1>  ,            _Alloc=std::allocator<std::pair<const std::string,std::string>>
1>          ]
1>          .../nonius/detail/argparse.h++(117) : see reference to class template instantiation 'std::unordered_map<std::string,std::string,std::hash<_Kty>,std::equal_to<_Kty>,std::allocator<std::pair<const _Kty,_Ty>>>' being compiled
1>          with
1>          [
1>              _Kty=std::string
1>  ,            _Ty=std::string
1>          ]
1>.../nonius/detail/stats.h++(131): warning C4244: 'initializing' : conversion from '__int64' to 'int', possible loss of data
1>          .../nonius/detail/stats.h++(209) : see reference to function template instantiation 'nonius::estimate<double> nonius::detail::bootstrap<Iterator,double(__cdecl *const &)(Iterator,Iterator)>(double,Iterator,Iterator,const nonius::detail::sample &,Estimator)' being compiled
1>          with
1>          [
1>              Iterator=std::_Vector_iterator<std::_Vector_val<std::_Simple_types<double>>>
1>  ,            Estimator=double (__cdecl *const &)(std::_Vector_iterator<std::_Vector_val<std::_Simple_types<double>>>,std::_Vector_iterator<std::_Vector_val<std::_Simple_types<double>>>)
1>          ]
1>.../nonius/detail/stats.h++(145): warning C4267: 'initializing' : conversion from 'size_t' to 'int', possible loss of data
1>.../nonius/detail/stats.h++(114): warning C4244: 'initializing' : conversion from '__int64' to 'int', possible loss of data
1>          .../nonius/detail/stats.h++(134) : see reference to function template instantiation 'nonius::detail::sample nonius::detail::jackknife<double(__cdecl *const &)(Iterator,Iterator),Iterator>(Estimator,Iterator,Iterator)' being compiled
1>          with
1>          [
1>              Iterator=std::_Vector_iterator<std::_Vector_val<std::_Simple_types<double>>>
1>  ,            Estimator=double (__cdecl *const &)(std::_Vector_iterator<std::_Vector_val<std::_Simple_types<double>>>,std::_Vector_iterator<std::_Vector_val<std::_Simple_types<double>>>)
1>          ]
1>          .../nonius/detail/stats.h++(209) : see reference to function template instantiation 'nonius::estimate<double> nonius::detail::bootstrap<Iterator,double(__cdecl *const &)(Iterator,Iterator)>(double,Iterator,Iterator,const nonius::detail::sample &,Estimator)' being compiled
1>          with
1>          [
1>              Iterator=std::_Vector_iterator<std::_Vector_val<std::_Simple_types<double>>>
1>  ,            Estimator=double (__cdecl *const &)(std::_Vector_iterator<std::_Vector_val<std::_Simple_types<double>>>,std::_Vector_iterator<std::_Vector_val<std::_Simple_types<double>>>)
1>          ]
```

Additional:

Please get also rid of the C-Style casts: e.g. (int)(variable)
use instead static_cast<int>(), That you can easily search for those casts and its also valid C++.

**Comment by rmartinho** [comment]
I'm currently in a transition period and don't have a machine around with VS2013 to check this out. If they're not too many, maybe you could fix them and send me a PR?

**Comment by rmartinho** [comment]
Fixed in 92281f6
